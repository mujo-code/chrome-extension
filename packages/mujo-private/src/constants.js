// config
export const API_HOST = 'https://getmujo.com'

// storage
export const SITE_TIME_KEY = 'SITE_TIME'
export const PREDICTED_BREAK_TIMES = 'PREDICTED_BREAK_TIMES'

// features
export const SMART_BREAKS_FEATURE = true

// alarms
export const SMART_BREAK = 'SMRT_BRK' // is only part of the key
// ex. SMRT_BRK_2019-08-07T22:17:13.246Z
