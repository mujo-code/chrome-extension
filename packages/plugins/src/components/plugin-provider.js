import React from 'react'

export const context = React.createContext()
const { Provider } = context

/*
  Plugin Provider
  -------
  This provider is going to accept some common tool inside of the
  extensions to allow the plugins to have access to common functionality
  within the app with out exposing it to window or something else gross

  env            - the environment in the extension (background | content | ntp)
  extension      - sugar sytax extension of browser api's
  storage        - access to the database
  config         - configuration for the extension
  model          - data structure for db
  constants      - constants variables defined in the extension

  Background only props

  changeEmitter  - storage updates propgated to hooks
  messageEmitter - event emmiter for messages
  alarms         - a way to listen to heartbeat ticks
  track          - a function to track with google analytics

  New tab page only props

  tabs           - some functions and state for tabs
  settings       - some functions and state for settings
  endScreen      - some functions to allow plugins to hook into the endscreen
*/

export const PluginProvider = ({
  env,
  extension,
  storage,
  config,
  changeEmitter,
  messageEmitter,
  constants,
  alarms,
  model,
  children,
  tabs,
  settings,
  isActive,
  track,
}) => (
  <Provider
    value={{
      env,
      extension,
      storage,
      config,
      changeEmitter,
      messageEmitter,
      constants,
      alarms,
      model,
      tabs,
      settings,
      isActive,
      track,
    }}
  >
    {children}
  </Provider>
)
