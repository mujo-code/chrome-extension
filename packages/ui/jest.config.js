const base = require('../../config/jest.base.js')

module.exports = {
  ...base,
}
