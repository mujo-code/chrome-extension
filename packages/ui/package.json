{
  "name": "@mujo/ui",
  "version": "2.3.0",
  "main": "dist/index.js",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/mujo-code/plugins.git"
  },
  "keywords": [
    "react"
  ],
  "files": [
    "dist/**/*"
  ],
  "author": "Jacob Lowe",
  "license": "MIT",
  "bugs": {
    "url": "https://github.com/mujo-code/plugins/issues"
  },
  "homepage": "https://github.com/mujo-code/plugins#readme",
  "scripts": {
    "test": "npm run lint && jest",
    "fmt": "prettier '{src,public}/**/*.js' --write",
    "lint": "eslint '{src,public}/**/*.js'",
    "build": "babel src --out-dir dist",
    "prepublish": "npm run build",
    "only-test": "jest --collectCoverage"
  },
  "devDependencies": {
    "@babel/cli": "^7.4.4",
    "@babel/core": "^7.4.5",
    "@babel/preset-env": "^7.4.5",
    "@babel/preset-react": "^7.0.0",
    "@babel/runtime": "^7.6.0",
    "@emotion/core": "^10.0.14",
    "@mujo/box": "^3.0.1",
    "@mujo/eslint-config": "^1.0.4",
    "@mujo/ingress": "^0.1.0",
    "@mujo/plugins": "^2.2.0",
    "@mujo/utils": "^2.3.0",
    "@testing-library/react": "^9.1.4",
    "@testing-library/react-hooks": "^3.1.0",
    "babel-plugin-transform-runtime": "^6.23.0",
    "eslint": "^5.16.0",
    "jest": "^24.9.0",
    "jest-environment-jsdom-fourteen": "1.0.1",
    "jest-image-snapshot": "^2.9.0",
    "jest-resolve": "24.9.0",
    "jest-watch-typeahead": "^0.3.1",
    "prettier": "^1.17.1",
    "react": "^16.9.0",
    "react-dom": "^16.9.0",
    "react-modal": "^3.10.1",
    "react-test-renderer": "^16.9.0"
  },
  "peerDependencies": {
    "@emotion/core": ">=10.0.14",
    "@mujo/box": "^3.0.1",
    "@mujo/ingress": ">=0.1.0",
    "@mujo/plugins": "*",
    "@mujo/utils": "*",
    "react": ">=16.8.6",
    "react-modal": ">=3.10.1"
  },
  "babel": {
    "presets": [
      "@babel/preset-env",
      "@babel/preset-react"
    ]
  },
  "browserslist": [
    "last 3 Chrome versions"
  ],
  "dependencies": {
    "@mujo/jest-webextension-mock": "^3.5.0",
    "@tweenjs/tween.js": "^18.3.1",
    "babel-jest": "^24.9.0",
    "react-feather": "^2.0.3",
    "react-spring": "^8.0.27"
  },
  "gitHead": "1a24f20513b5e7fa7d2e3d0245142cc61a4ae337"
}
