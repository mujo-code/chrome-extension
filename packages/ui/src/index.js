export * from './components'
export * from './styles/colors'
export * from './hooks'
