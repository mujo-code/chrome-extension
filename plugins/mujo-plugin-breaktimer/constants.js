export const ALARM_KEY = 'MINDFUL_ALARM'
export const PREDICTED_TIMES_FEATURE = 'PREDICTED_BREAK_TIMES_FEATURE'
export const HOUR_HALF = 1000 * 60 * 60 * 1.5
