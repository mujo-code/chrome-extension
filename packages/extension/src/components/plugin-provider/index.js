import { PluginProvider as Provider } from '@mujo/plugins'
import { Extension } from '@mujo/utils'
import React from 'react'
import * as constants from '../../constants'
import { useExtension } from '../../hooks/use-extension'
import { i18n } from '../../i18n'
import model from '../../model'

// client plugin providers
export const NTPProvider = props => {
  const {
    pushTab,
    currentTab,
    removeTab,
    pushSetting,
    removeSetting,
    updateSetting,
    registerEndScreen,
    removeEndScreen,
  } = useExtension()
  const tabs = { pushTab, removeTab, currentTab }
  const endScreen = {
    registerEndScreen,
    removeEndScreen,
  }
  const settings = { pushSetting, removeSetting, updateSetting }
  return (
    <Provider
      {...props}
      tabs={tabs}
      settings={settings}
      endScreen={endScreen}
    />
  )
}

export const PluginProvider = ({ children, env }) => {
  const props = {
    env,
    extension: {
      ...Extension,
      i18n: { ...Extension.i18n, t: i18n.t.bind(i18n) },
    },
    constants,
    tabs: {},
    settings: {},
    model,
  }
  // add additional info for ntp
  return env === 'ntp' ? (
    <NTPProvider {...props}>{children}</NTPProvider>
  ) : (
    <Provider {...props}>{children}</Provider>
  )
}
