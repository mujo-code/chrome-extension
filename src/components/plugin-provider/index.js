import { PluginProvider as Provider } from '@mujo/plugins'
import React from 'react'
import * as constants from '../../constants'
import { useExtension } from '../../hooks/use-extension'
import * as extension from '../../lib/extension'

// NOTE we use a proxy to allow for access to `useExtension`
export const NTPPluginProvider = ({ children }) => {
  const {
    pushTab,
    currentTab,
    removeTab,
    pushSetting,
    removeSetting,
    updateSetting,
  } = useExtension()
  const tabs = { pushTab, removeTab, currentTab }
  const settings = { pushSetting, removeSetting, updateSetting }
  const props = {
    env: 'ntp',
    extension,
    constants,
    tabs,
    settings,
  }
  return <Provider {...props}>{children}</Provider>
}
