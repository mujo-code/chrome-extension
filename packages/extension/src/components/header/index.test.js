import { Icon, BreathPlayer } from '@mujo/ui'
import React from 'react'
import { create } from 'react-test-renderer'
import { SETTINGS_MODAL } from '../../constants'
import { PluginProvider } from '../plugin-provider'
import { Header } from '.'

test('Header component matches snapshot', () => {
  const tree = create(
    <PluginProvider>
      <Header />
    </PluginProvider>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('Header component matches open snapshot', () => {
  const tree = create(
    <PluginProvider>
      <Header playerIsOpen={true} />
    </PluginProvider>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

test('Header should open settings modal when the Icon is clicked', () => {
  const fn = jest.fn()
  const wrapper = create(
    <PluginProvider>
      <Header setUpsellModal={fn} />
    </PluginProvider>
  )
  const { root } = wrapper
  const icon = root.findByType(Icon)

  // mock click on modal
  icon.props.onClick({ target: icon })
  expect(fn).toBeCalledWith({ name: SETTINGS_MODAL })
})

test('Header should open settings modal when the Icon is clicked', () => {
  const fn = jest.fn()
  const wrapper = create(
    <PluginProvider>
      <Header setUpsellModal={fn} />
    </PluginProvider>
  )
  const { root } = wrapper
  const icon = root.findByType(Icon)

  // mock click on modal
  icon.props.onClick({ target: icon })
  expect(fn).toBeCalledWith({ name: SETTINGS_MODAL })
})

test('Clicking the Header player should open it', () => {
  const fn = jest.fn()
  const wrapper = create(
    <PluginProvider>
      <Header setPlayerIsOpen={fn} />
    </PluginProvider>
  )
  const { root } = wrapper
  const player = root.findByType(BreathPlayer)

  // mock click on modal
  player.props.onClick({ target: player })
  expect(fn).toBeCalledWith(true)
})
