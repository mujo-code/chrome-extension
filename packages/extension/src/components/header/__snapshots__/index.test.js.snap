// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component matches open snapshot 1`] = `
<div
  className="css-1s1haim"
  layer="2"
  position="relative"
>
  <div
    className="css-4rodaq"
    onClick={[Function]}
  >
    <svg
      fill="none"
      height={32}
      icon="menu"
      stroke="#EAE2EB"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      viewBox="0 0 24 24"
      width={32}
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="3"
        x2="21"
        y1="12"
        y2="12"
      />
      <line
        x1="3"
        x2="21"
        y1="6"
        y2="6"
      />
      <line
        x1="3"
        x2="21"
        y1="18"
        y2="18"
      />
    </svg>
  </div>
  <div
    className="css-1wl0vcq"
  >
    <div
      className="css-97bura"
      data-testid="breath-player"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="css-1uugaji"
        style={
          Object {
            "borderRadius": "30vh",
            "height": "30vh",
            "left": "50%",
            "marginLeft": "-15vh",
            "marginTop": "-15vh",
            "top": "50%",
            "transform": "scale(1)",
            "transition": "width 0.7s ease-in-out,height 0.7s ease-in-out,margin-left 0.7s ease-in-out,margin-top 0.7s ease-in-out,border-radius 0.7s ease-in-out",
            "width": "30vh",
          }
        }
      />
      <div
        className="css-1o44dsb"
        style={
          Object {
            "borderRadius": "30vh",
            "height": "30vh",
            "left": "50%",
            "marginLeft": "-15vh",
            "marginTop": "-15vh",
            "top": "50%",
            "transform": "scale(1)",
            "transition": "width 0.7s ease-in-out,height 0.7s ease-in-out,margin-left 0.7s ease-in-out,margin-top 0.7s ease-in-out,border-radius 0.7s ease-in-out",
            "width": "30vh",
          }
        }
      />
      <div
        className="css-ykw1nj"
        style={Object {}}
      >
        <h2
          className="css-j5os12"
        >
          Breathe in
        </h2>
      </div>
      <div
        className="css-ykw1nj"
        style={
          Object {
            "opacity": 0,
            "transform": "scale(1.1)",
          }
        }
      >
        <h2
          className="css-j5os12"
        >
          Breathe out
        </h2>
      </div>
      <h4
        className="css-1r3e8hh"
      >
        5
      </h4>
      <div
        className="css-cejex2"
        onClick={[Function]}
        style={
          Object {
            "opacity": 0,
            "transform": "scale(1.1)",
          }
        }
      >
        <div
          className="css-1rr4qq7"
        />
        <div
          className="css-1rr4qq7"
        />
      </div>
    </div>
  </div>
  <div
    className="css-79elbk"
  >
    <div
      className="css-4pw2fe"
    >
      <span
        className="css-xrmt9z"
      >
        <div
          className="css-1as2nky"
        />
        <div
          className="css-91tplw"
        >
          Take a break
        </div>
      </span>
    </div>
  </div>
</div>
`;

exports[`Header component matches snapshot 1`] = `
<div
  className="css-1s1haim"
  layer="2"
  position="relative"
>
  <div
    className="css-1kyoofh"
    onClick={[Function]}
  >
    <svg
      fill="none"
      height={32}
      icon="menu"
      stroke="#EAE2EB"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      viewBox="0 0 24 24"
      width={32}
      xmlns="http://www.w3.org/2000/svg"
    >
      <line
        x1="3"
        x2="21"
        y1="12"
        y2="12"
      />
      <line
        x1="3"
        x2="21"
        y1="6"
        y2="6"
      />
      <line
        x1="3"
        x2="21"
        y1="18"
        y2="18"
      />
    </svg>
  </div>
  <div
    className="css-13u4gsy"
  >
    <div
      className="css-1y63kkb"
      data-testid="breath-player"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      <div
        className="css-1uugaji"
        style={
          Object {
            "borderRadius": 80,
            "height": 40,
            "left": "50%",
            "marginLeft": -20,
            "marginTop": -20,
            "top": "50%",
            "transform": "scale(1)",
            "transition": "width 0.7s ease-in-out,height 0.7s ease-in-out,margin-left 0.7s ease-in-out,margin-top 0.7s ease-in-out,border-radius 0.7s ease-in-out",
            "width": 40,
          }
        }
      />
      <div
        className="css-1o44dsb"
        style={
          Object {
            "borderRadius": 80,
            "height": 40,
            "left": "50%",
            "marginLeft": -20,
            "marginTop": -20,
            "top": "50%",
            "transform": "scale(1)",
            "transition": "width 0.7s ease-in-out,height 0.7s ease-in-out,margin-left 0.7s ease-in-out,margin-top 0.7s ease-in-out,border-radius 0.7s ease-in-out",
            "width": 40,
          }
        }
      />
    </div>
  </div>
  <div
    className="css-79elbk"
  >
    <div
      className="css-4pw2fe"
    >
      <span
        className="css-xrmt9z"
      >
        <div
          className="css-1as2nky"
        />
        <div
          className="css-91tplw"
        >
          Take a break
        </div>
      </span>
    </div>
  </div>
</div>
`;
